File: Practice/atcoder/ABC/070/src/c.py
# まあ最小公倍数lcmでしょう lcmのアルゴリズムはしらんけどたしかgcdを求めてからなんかするやつ

n = int(input())
x = int(input())


def gcd(x, y):
    if x > y:
        return gcd(y, x)
    while x != 0:
        x, y = y % x, x
    return y


def lcm(x, y):
    return (x*y)//gcd(x, y)


for _ in range(n - 1):
    y = int(input())
    x = lcm(x, y)

print(x)

File: Practice/atcoder/ABC/070/src/c.go
// gcd lcm毎回勘で書いてるよね
// py, rbで解いてたから気づかんかったけどオーバーフローするなこれ

package main

import "fmt"

func main() {
	var N int
	fmt.Scan(&N)
	var ans int
	fmt.Scan(&ans)
	for i := 1; i < N; i++ {
		var V int
		fmt.Scan(&V)
		ans = lcm(ans, V)
	}
	fmt.Println(ans)
}

func lcm(a, b int) int {
	return a / gcd(a, b) * b
}

func gcd(a, b int) int {
	if a > b {
		a, b = b, a
	}
	for a > 0 {
		a, b = b%a, a
	}
	return b
}
